community/kernel/src/main/java/org/neo4j/kernel/impl/store/TemporalType.java

```diff
@@ -77,9 +89,22 @@ public int calculateNumberOfBlocksUsedForTemporal( long firstBlock )
                 }
 
                 @Override
-                public ArrayValue decodeArray( TemporalHeader header, byte[] data )
+                public ArrayValue decodeArray( Value dataValue )
                 {
-                    throw new UnsupportedOperationException( "Add support for arrays" );
+                    if ( dataValue instanceof LongArray )
+                    {
+                        LongArray numbers = (LongArray) dataValue;
+                        LocalDate[] dates = new LocalDate[numbers.length()];
+                        for ( int i = 0; i < dates.length; i++ )
+                        {
+                            dates[i] = LocalDate.ofEpochDay( numbers.longValue( i ) );
+                        }
+                        return Values.dateArray( dates );
+                    }
+                    else
+                    {
+                        throw new RuntimeException( "Array with unexpected type. Actual:" + dataValue.getClass().getSimpleName() + ". Expected: LongArray." );
```

Comment:
Would be better to throw some more specific and coherent exception, like `InvalidRecordException` or the likes. Likewise for the other similar places.